crates/proto/src/domain/batch.rs

```diff
@@ -0,0 +1,53 @@
+use std::collections::BTreeMap;
+
+use miden_objects::{
+    block::BlockHeader,
+    note::{NoteId, NoteInclusionProof},
+    transaction::ChainMmr,
+    utils::{Deserializable, Serializable},
+};
+
+use crate::{
+    errors::{ConversionError, MissingFieldHelper},
+    generated::responses::GetBatchInputsResponse,
```

Comment:
nit: I would import this as `generated::responses as proto`. This way, we'd write below:
```rust
impl From<BatchInputs> for proto::GetBatchInputsResponse
```
This makes it a bit more clear what's what (and this is the convention we've tried to use in other places as well).